src/main/java/gregtech/integration/jei/GTJeiPlugin.java

```diff
@@ -132,13 +139,11 @@ public void register(IModRegistry registry) {
                     if (metaTileEntity instanceof SteamMetaTileEntity) {
                         deferredCatalysts.add(metaTileEntity);
                     } else if (metaTileEntity instanceof IMultipleRecipeMaps && ((IMultipleRecipeMaps) metaTileEntity).hasMultipleRecipeMaps()) {
-                        multipleRecipeMapCatalysts.add(metaTileEntity);
-                    } else {
-                        RecipeMap<?> recipeMap = ((AbstractRecipeLogic) workableCapability).getRecipeMap();
-                        registry.addRecipeCatalyst(metaTileEntity.getStackForm(), GTValues.MODID + ":" + recipeMap.unlocalizedName);
-                        if (recipeMap instanceof RecipeMapFurnace) {
-                            registry.addRecipeCatalyst(metaTileEntity.getStackForm(), VanillaRecipeCategoryUid.SMELTING);
+                        for (RecipeMap<?> recipeMap : ((IMultipleRecipeMaps) metaTileEntity).getAvailableRecipeMaps()) {
+                            registerRecipeMapCatalyst(registry, recipeMap, metaTileEntity);
                         }
+                    } else {
+                        registerRecipeMapCatalyst(registry, ((AbstractRecipeLogic) workableCapability).getRecipeMap(), metaTileEntity);
```

Comment:
Would this have issues for things like the LCE and regular combustion gens, if we decided to make some recipes that are LCE only?